### ISSUE TITLE
Processing model data for sterodynamics module

### ISSUE THREAD

**Issue by jetesdal** [radical-collaboration/facts#322]
I came across some **zos** fields in `tlm_sterodynamics_preprocess_data.tgz` that are not matching with the version I downloaded from the CMIP6 archive. I am not sure if it is because the zos field was updated in the archive or the preprocessing step (e.g., de-drifting or re-gridding to 1x1) have created the mismatch.

Below is an example showing the linear trend of **zos** for the period 1993 to 2020 using the historical and the ssp585 **zos** files found in `tlm_sterodynamics_preprocess_data.tgz`. You can see that the trend for UKESM1-0-LL look unrealistic.
![figure (8)](https://github.com/radical-collaboration/facts/assets/8991018/8f816b8a-eb17-4615-8762-2d1cf2be0452)

In this case, the underlying reason is that the values between the historical and ssp585 dataset do not align well, as seen from the time series of the global mean of **zos**:
![figure (9)](https://github.com/radical-collaboration/facts/assets/8991018/091be9ee-bb40-4642-84c7-29e7c53a525b)

These artifacts are not present in the recently downloaded CMIP6 **zos** output, which suggests they might have been introduced during the preprocessing stage. Are there any scripts and/or documentation available for the CMIP6 processing involved in creating the input data in `tlm_sterodynamics_cmip6_data.tgz`?

I think having the scripts and tools would also help in addressing issue #129 and would allow a user to update the input data with more recent CMIP6 output or including post-CMIP6 model output.

**Comment by bobkopp** [comment]
@Timh37 handled this -- Tim, could you please address?

**Comment by bobkopp** [comment]
Also please check to see if underlying files on ESGF have been updated since 9/2020. It's quite possible that the difference you're seeing reflects file changes.

**Comment by bobkopp** [comment]
Also, is UKESM the only example of this? As you can see in Table A3 of the FACTS manuscript, UKESM was not used in the calibration of the emulator -- perhaps because the version on ESGF at the time was flaky.

**Comment by jetesdal** [comment]
Thank you for the clarification and the insights @bobkopp. I wasn't aware that the IBE correction was already included in the zos dataset. Yes, it looks like the IBE (i.e. zos+psl) explains the issue with the UKESM1-0-LL.

I think the very first step in addressing #129 and #171 (and I presume also #252 ?)  would be to demonstrate reproducibility of the current module-data version. The current set of maps I gathered are from FACTS module-data and I added a set of GFDL models.
![figure (13)](https://github.com/radical-collaboration/facts/assets/8991018/3f5c9d4a-202a-4e08-a8c6-8811d69cc2bf)

I figured out a way to do the dedrifting and regridding for the additional model output, but I would need to also do the IBE corrections. It would be great to have a detailed account on the preprocessing steps applied to the CMIP6 output for the existing module-data version. @Timh37, is that something you could share?

**Comment by Timh37** [comment]
The preprocessing steps are detailed in the SM of Chapter 9 (including the exclusion of UKESM1-0-LL. I haven't uploaded the preprocessing scripts in a central place yet but bits and pieces are available [here](https://github.com/Timh37/gmslCMIP6/tree/master/preproc) and [here](https://github.com/Timh37/SeasonalDSLC_NWES/tree/main/cmip6_processing). I'll get back with the IBE script which is not included in these repositories. It would be nice to integrate all of these into FACTS at some point but I don't really have time for this at the moment.

**Comment by bobkopp** [comment]
@jetesdal I've put the code from @Timh37 here: https://github.com/radical-collaboration/facts/tree/development/modules/tlm/sterodynamics/2lmfit 

I think we've solved the original question, given the known issue with the UKESM IBE correction, but let me know if there are other bits and pieces you need that aren't in the repo now.

**Comment by jetesdal** [comment]
Yes, I've got all the info I was looking for. Thank you very much for your help @bobkopp and @Timh37 for the clarifications and pointing me to all the sources.

I went ahead and applied the IBE correction for a subset of models and it looks like that the UKESM1-0-LL is now giving a realistic IBE with the latest `psl` dataset downloaded from the CMIP6 archive.
![figure (3)](https://github.com/radical-collaboration/facts/assets/8991018/25744e7b-1132-4e03-a7d4-78c7d5f20bc2)
![figure (4)](https://github.com/radical-collaboration/facts/assets/8991018/7ad54a1b-ee51-484b-ad27-b1e842743632)
